### PR TITLE
[GEP-13] Make ManagedSeed spec.shoot field optional

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Makes the `ManagedSeed` `spec.shoot` field optional to avoid having to specify it in a `ManagedSeedSet` template.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/registry/seedmanagement/managedseed/strategy.go [hunk 1]
```diff
@@ -165,5 +165,12 @@ func ShootNameTriggerFunc(obj runtime.Object) string {
 		return ""
 	}
 
+	return GetShootName(managedSeed)
+}
+
+func GetShootName(managedSeed *seedmanagement.ManagedSeed) string {
```
Comment (rfranzke) [inline pkg/registry/seedmanagement/managedseed/strategy.go:172]:
```suggestion
func getShootName(managedSeed *seedmanagement.ManagedSeed) string {
```

Comment (stoyanr) [reply pkg/registry/seedmanagement/managedseed/strategy.go:172]: It's used outside of this package, see https://github.com/stoyanr/gardener/blob/34bef2b51fa1c8083aa01ab22227353ada296259/pkg/registry/seedmanagement/managedseed/storage/tableconvertor.go#L87

Comment (rfranzke) [reply pkg/registry/seedmanagement/managedseed/strategy.go:172]: OK, missed that. Then I would only suggest to add a doc string as it's exported. Rest lgtm.

Comment (stoyanr) [reply pkg/registry/seedmanagement/managedseed/strategy.go:172]: Thanks, done.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**